ablateLibrary/monitors/ignitionDelayPeakYi.cpp

```diff
@@ -0,0 +1,146 @@
+    : log(logIn ? logIn : std::make_shared<logs::StdOut>()), historyLog(historyLogIn), species(species), location(location) {}
+
+ablate::monitors::IgnitionDelayPeakYi::~IgnitionDelayPeakYi() {
+    // compute the time at the maximum yi
+    std::size_t loc = 0;
+    double maxValue = 0.0;
+    for (std::size_t i = 0; i < yiHistory.size(); i++) {
+        if (yiHistory[i] > maxValue) {
+            maxValue = yiHistory[i];
+            loc = i;
+        }
+    }
+
+    log->Printf("Computed Ignition Delay (%s): %g\n", species.c_str(), timeHistory[loc]);
+}
+
+void ablate::monitors::IgnitionDelayPeakYi::Register(std::shared_ptr<Monitorable> monitorableObject) {
+    // this probe will only work with fV flow with a single mpi rank for now.  It should be replaced with DMInterpolationEvaluate
+    auto flow = std::dynamic_pointer_cast<ablate::flow::FVFlow>(monitorableObject);
+    if (!flow) {
+        throw std::invalid_argument("The IgnitionDelay monitor can only be used with ablate::flow::FVFlow");
+    }
+
+    // check the size
+    int size;
+    MPI_Comm_size(PetscObjectComm((PetscObject)flow->GetDM()), &size) >> checkMpiError;
+    if (size != 1) {
+        throw std::runtime_error("The IgnitionDelay monitor only works with a single mpi rank");
+    }
+
+    // determine the component offset
+    auto eulerIdValue = flow->GetFieldId("euler");
+    if (eulerIdValue) {
+        eulerId = eulerIdValue.value();
+    } else {
+        throw std::invalid_argument("The IgnitionDelay monitor expects to find euler in the ablate::flow::FVFlow");
+    }
+
+    auto densityYiIdValue = flow->GetFieldId("densityYi");
+    if (densityYiIdValue) {
+        yiId = densityYiIdValue.value();
+    } else {
+        throw std::invalid_argument("The IgnitionDelay monitor expects to find densityYi in the ablate::flow::FVFlow");
+    }
+
+    const auto& speciesList = flow->GetFieldDescriptor("densityYi").componentNames;
+    yiOffset = -1;
+    for (std::size_t sp = 0; sp < speciesList.size(); sp++) {
+        if (speciesList[sp] == species) {
+            yiOffset = sp;
+        }
+    }
+    if (yiOffset < 0) {
+        throw std::invalid_argument("The IgnitionDelay monitor cannot find the " + species + " species");
+    }
+
+    // Locate the closest cell
+    PetscReal distance = PETSC_MAX_REAL;
+
+    // Get the cell start and end for the fv cells
+    PetscInt cellStart, cellEnd;
+    DMPlexGetHeightStratum(flow->GetDM(), 0, &cellStart, &cellEnd) >> checkError;
+
+    // Extract the cell geometry, and the dm that holds the information
+    Vec cellGeomVec;
+    DM dmCell;
+    const PetscScalar* cellGeomArray;
+    DMPlexGetGeometryFVM(flow->GetDM(), NULL, &cellGeomVec, NULL) >> checkError;
+    VecGetDM(cellGeomVec, &dmCell) >> checkError;
+    VecGetArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
+
+    for (PetscInt c = cellStart; c < cellEnd; ++c) {
```

Comment:
Note that you might be able to use https://petsc.org/release/docs/manualpages/DM/DMLocatePoints.html

Reply:
Thanks!  I figured there was a better way.  I've updated the PR with the changes.